sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -1,2 +1,67 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
+use crate::{
+    models::BlobProperties, pipeline::StorageHeadersPolicy, BlobBlobClientGetPropertiesOptions,
+    BlobClientOptions, GeneratedBlobClient,
+};
+use azure_core::{credentials::TokenCredential, BearerTokenCredentialPolicy, Policy, Result};
+use std::sync::Arc;
+
+pub struct BlobClient {
+    pub endpoint: String,
+    pub container_name: String,
+    pub blob_name: String,
+    client: GeneratedBlobClient,
+}
+
+impl BlobClient {
+    pub fn new(
+        endpoint: String,
```

Comment:
Should take a `&str`. See generated clients. You'd then `let endpoint: Url = endpoint.parse()?;`.